### ISSUE TITLE
Can we add a boot animation?

### ISSUE THREAD

**Issue by S1700** [jcubic/jquery.terminal#782]
So if you use Linux you know that whenever you boot into the OS that you use you may see a bunch of text on the screen that looks like this:
![image](https://user-images.githubusercontent.com/68336568/169873633-f89c27b4-5b96-4a39-ba41-0fd891811785.png)
Is it possible to do something like that but every time you go onto the website it does a fake boot/loading screen that looks like that? I will try adding some cookies too so the users won't get it all the time because it gets annoying.

Thanks

**Comment by jcubic** [comment]
It's just a bunch of echo you can create something like this with no issue. You don't need animation for this. Unless you want animation of progress bars like with `apt install` on ubuntu.

This can be coded like this:

```javascript
const sequence = ['Line One', 'Line Two']; // put all lines here.

var term = $('body').terminal({}, {
    onInit() {
        const term = this;
        if (!localStorage.getItem('visited')) {
            term.pause(true);
            (function loop() {
                term.echo(sequence.shift());
                if (sequence.length) {
                    setTimeout(loop, 500);
                } else {
                    localStorage.setItem('visited', 'true');
                    term.resume();
                }
            })();
        }
    }
});
```

You only need to provide the strings that will be printed.

**Comment by S1700** [comment]
oh didn't think of it like that thanks!

**Comment by S1700** [comment]
Sorry to carry this on but I have a slight issue, I've added that code to mine but it's not running, no errors, just not working. Here are some screenshots:
![image](https://user-images.githubusercontent.com/68336568/169906509-e07ed037-b4f9-4260-bf7b-6394193d8399.png)
And the cookie its even set:
![image](https://user-images.githubusercontent.com/68336568/169906608-04f6c5e8-e670-4bd5-a246-26208f3a311f.png)

Thanks

**Comment by dragoncoder047** [comment]
The reason the cookie is not being set is because @jcubic's code uses `localStorage` and not cookies.

![Screen Shot 2022-05-23 at 5 22 31 PM](https://user-images.githubusercontent.com/101021094/169908260-dfee5c1d-fb74-4ea6-9d76-8dee1a7eaeee.png)

**Comment by S1700** [comment]
Ah i see, unfortunately its still not there
![image](https://user-images.githubusercontent.com/68336568/169908525-43340030-48d2-4344-8a61-6650be99ae98.png)
Could running this locally cause this, just wanna make sure

**Comment by dragoncoder047** [comment]
Usually, `file:///` urls don't get special privileges so that may be your problem.

FYI, the code up on https://s1700.github.io/ is is not the code in [your comment](https://github.com/jcubic/jquery.terminal/issues/782#issuecomment-1135142116). Might want to push those changes so they are live.

**Comment by S1700** [comment]
alr ill push now gimme a sec

**Comment by S1700** [comment]
okay, I've pushed and all live. What do you mean by `file://`

**Comment by S1700** [comment]
should i just try using cookies?

**Comment by dragoncoder047** [comment]
No, localStorage actually works better because it has a simple API whereas with cookies you have to parse the `document.cookie` string yourself.

`file://` urls are what you get if you right click the .html file and select "open in &lt;your web browser>". They have lowered privileges than even `http://` (non-secure) URLS because lots of phishing emails try to trick you into opening a .html file.

Anyway, your changes are live, and they don't work for some reason. All I can think of is you have a `greetings` **and** an `onInit`. @jcubic can you confirm if `greetings` overrides `onInit`?

**Comment by jcubic** [comment]
You have an error in your code. This is not what my code looks like you have command onInit, instead of option onInit.

**Comment by S1700** [comment]
ahhh okay ill try that thanks

**Comment by S1700** [comment]
Works, thanks so much. I'll also try implementing random delay time to make it feel more real

Thanks

**Comment by jcubic** [comment]
You have setTimeout you can use a random (Math.random return numbers if 0-1) value instead of one second 1000. See: [Generate random number between two numbers in JavaScript](https://stackoverflow.com/q/4959975/387194).
